lib/explorer/shared.ex

```diff
@@ -99,8 +100,17 @@ defmodule Explorer.Shared do
     {:naive_datetime, precision}
   end
 
-  def normalise_dtype({:decimal, _precision, _scale} = dtype), do: dtype
-  def normalise_dtype(:decimal), do: {:decimal, nil, 2}
+  def normalise_dtype({:decimal, precision, scale} = dtype)
+      when is_integer(scale) and (is_nil(precision) or is_integer(precision)),
+      do: dtype
+
+  def normalise_dtype(:decimal), do: {:decimal, nil, 0}
+  def normalise_dtype(:d0), do: {:decimal, nil, 0}
+  def normalise_dtype(:d1), do: {:decimal, nil, 1}
+  def normalise_dtype(:d2), do: {:decimal, nil, 2}
+  def normalise_dtype(:d3), do: {:decimal, nil, 3}
+  def normalise_dtype(:d4), do: {:decimal, nil, 4}
+  def normalise_dtype(:d5), do: {:decimal, nil, 5}
```

Comment:
```suggestion
  for scale <- 0..38 do
    def normalise_dtype(unquote(:"d#{scale}")), do: {:decimal, nil, unquote(scale)}
  end
```

If we wanted to go all the way up to 38. If not pls ignore :)